src/net.rs

```diff
@@ -31,6 +31,22 @@ pub trait NetworkListener: Clone {
     fn incoming(&mut self) -> NetworkConnections<Self> {
         NetworkConnections(self)
     }
+
+    /// Sets the read timeout for all streams that are accepted
+    fn set_read_timeout(&mut self, _: Option<Duration>) {
+        // This default implementation is only here to prevent the addition of
+        // these methods from being a breaking change. They should be removed
+        // when the next breaking release is made.
+        unimplemented!();
```

Comment:
You're right. I didn't think this through.

I think a `warn!` would be appropriate. Doing nothing would just mean that timeouts would be silently ignored, which could be very surprising to both users and implementers of custom `NetworkConnector`s. An `error!` seems too strong to me, as nothing is actually going wrong (yet) at this point.